### ISSUE TITLE
Fix error to get descomposition

### ISSUE THREAD

**Issue by joseplz** [artiste-qb-net/qubiter#18]
Is is it possible to fix an error in the QSD compiler and get the approximated matrix? 
I want to study how many gates i need to produce a unitary operation in function of the error.

**Comment by rrtucci** [comment]
Wat is the error joseplz?

**Comment by rrtucci** [comment]
Oh, sorry, I misunderstood. You mean, fix a parameter epsilon so that norm(u - approx_u) < epsilon

**Comment by joseplz** [comment]
Exactly. Like an implementation of the Solovay Kitaev theorem. I ve been reading the jupyternotebooks, and as i have seen, you can get de descompostion and later the error. But what i want is to fix the error and then get the approximation.

**Comment by rrtucci** [comment]
It is a very good request, but will take a lot of time and work. I promise to do it someday...

**Comment by joseplz** [comment]
I'll be very grateful if you can. BTW, wich matrix norm is used to calculate the error?

**Comment by rrtucci** [comment]
I always use the 2 norm, aka the Frobenius norm, but I am not very sophisticated about norms

**Comment by rrtucci** [comment]
The multiplexors decpose exactly into a huge number of gates but they can be approximated. I have included software to do a possible approximation of them. But it requires ancillary qubits

**Comment by rrtucci** [comment]
I think multiplexors and their approximation will be important in quantum AI

**Comment by joseplz** [comment]
Well im interested on problems related to quantum chemestry. I want to study quantum circuits that produces the ground state of some hamiltonian, and then relate the error of the circuit respecto to the ideal with some parameters.
